### PR TITLE
Backend/mutation

### PR DESCRIPTION
**Pull request by jeffreyhelgers**
Implemented the first two mutations, insertion and deletion. The algorithm that computes the mutations is brute force. All the nodes are compared to each other. If you select a specific genome, you can use the method 'getGenomeMutations', with the genome id as its parameter, to get all the mutations on this genome.

### CODE REVIEW

**Comment by MatthijsRijlaarsdam** [comment]
An insertion is a deletion viewed from the other genome (when comparing two Genomes). Why make the distinction between the two? Shouldn't we just have one InDel class?

**Comment by jeffreyhelgers** [comment]
True,  but when you want to compare one genemo with another genome you see all the mutations according to the base genome.  If we have one InDel class you have to give the objects for instance a boolean to say that a deletion or insertion appeared on the base genome.

**Comment by jeffreyhelgers** [comment]
True,  but when you want to compare one genemo with another genome you see all the mutations according to the base genome.  If we have one InDel class you have to give the objects for instance a boolean to say that a deletion or insertion appeared on the base genome.

**Comment by MatthijsRijlaarsdam** [comment]
Right, we're implementing a tree structure where we'll store the mutations in, something like:

Level 1: nodes 1,6,7
Level 2: 2,3,4-> insertion at 2 of (2,3) | deletion at 2 of 4. (so 4 is not at the root in level 1)
Level 3: genome A  | Genome B

Compare A and B:
     ..........  (INSERTION:2    3)   4 
    ......  /      ................................ \
**1**      ............................................     **6** ............   **7**
      ...... \    .................................. /
      .........      (DELETION 4)

File: backEnd/src/main/java/mutation/AllMutations.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+		
+	/**
+	 * Constructor to create.
+	 * @param dc The given data container.
+	 */
+	public AllMutations(DataContainer dc) {
+		this.dc = dc;
+		mutations = new HashMap<>();
+		setKeys(dc.getGenomes().keySet());
+		checkMutations();
```
Comment (MatthijsRijlaarsdam) [inline backEnd/src/main/java/mutation/AllMutations.java]: comment these methods please

**Comment by jeffreyhelgers** [comment]
All the methods are already provided with javadoc.

**Pull request merged**